envoy/network/connection.h

```diff
@@ -195,7 +195,7 @@ class Connection : public Event::DeferredDeletable,
    * @return the address provider backing this connection.
    */
   virtual const SocketAddressProvider& addressProvider() const PURE;
-  virtual SocketAddressProviderSharedPtr addressProviderSharedPtr() const PURE;
+  virtual SocketAddressSetterSharedPtr addressProviderSharedPtr() const PURE;
```

Comment:
note: I have to change this due to allow the StreamInfo to set the upstream SSL connection into the AddressProvider.